indexing-hadoop/src/main/java/io/druid/indexer/updater/HadoopConverterJob.java

```diff
@@ -0,0 +1,1021 @@
+          log.info("Committing new segment");
+          taskContext.progress();
+
+          final FileSystem fs = taskContext.getWorkingDirectory().getFileSystem(taskContext.getConfiguration());
+          final Path taskAttemptDir = getTaskPath(
+              context.getJobID(),
+              context.getTaskAttemptID(),
+              taskContext.getWorkingDirectory()
+          );
+          final Path taskAttemptFile = new Path(taskAttemptDir, DATA_FILE_KEY);
+          final Path taskAttemptSuccess = new Path(taskAttemptDir, DATA_SUCCESS_KEY);
+          try (final OutputStream outputStream = fs.create(taskAttemptFile, false, 1 << 10, commitProgressable)) {
+            outputStream.write(HadoopDruidConverterConfig.jsonMapper.writeValueAsBytes(newSegment));
+          }
+
+          fs.create(taskAttemptSuccess, false).close();
+
+          taskContext.progress();
+          taskContext.setStatus("Committed");
+        }
+
+        @Override
+        public void abortTask(TaskAttemptContext taskContext) throws IOException
+        {
+          log.warn("Aborting task");
+        }
+      };
+    }
+  }
+
+
+  public static class ConvertingMapper extends Mapper<String, String, Text, Text>
+  {
+    private static final Logger log = new Logger(ConvertingMapper.class);
+    private static final String TMP_FILE_LOC_KEY = "io.druid.indexer.updater.converter.reducer.tmpDir";
+
+    @Override
+    protected void map(
+        String key, String value,
+        final Context context
+    ) throws IOException, InterruptedException
+    {
+      final InputSplit split = context.getInputSplit();
+      if (!(split instanceof DataSegmentSplit)) {
+        throw new IOException(
+            String.format(
+                "Unexpected split type. Expected [%s] was [%s]",
+                DataSegmentSplit.class.getCanonicalName(),
+                split.getClass().getCanonicalName()
+            )
+        );
+      }
+
+      final String tmpDirLoc = context.getConfiguration().get(TMP_FILE_LOC_KEY);
+      final File tmpDir = Paths.get(tmpDirLoc).toFile();
+
+      final DataSegment segment = ((DataSegmentSplit) split).getDataSegment();
+
+      final HadoopDruidConverterConfig config = converterConfigFromConfiguration(context.getConfiguration());
+
+      context.setStatus("DOWNLOADING");
+      context.progress();
+      final File inDir = new File(tmpDir, "in");
+      if (inDir.exists() && !inDir.delete()) {
+        log.warn("Could not delete [%s]", inDir);
+      }
+      if (!inDir.mkdir()) {
+        log.warn("Unable to make directory");
+      }
+      final File tmpDownloadFile = File.createTempFile("dataSegment", ".zip");
+      if (tmpDownloadFile.exists() && !tmpDownloadFile.delete()) {
+        log.warn("Couldn't clear out temporary file [%s]", tmpDownloadFile);
+      }
+      try {
+        final Path inPath = new Path(getURIFromSegment(segment));
+        final FileSystem fs = inPath.getFileSystem(context.getConfiguration());
```

Comment:
The `serializeOutIndex` job for the `IndexGeneratorJob` is not pluggable at all. It is deeply interweaved with the IndexGeneratorJob way of doing things including the structure of its config.

It would be worth making that codepath more modular in general, but that feels like a pretty big scope creep for this PR unless it really is required.